src/twisted/conch/ssh/transport.py

```diff
@@ -637,83 +659,186 @@ def sendPacket(self, messageType, payload):
             payload = self.outgoingCompression.compress(
                 payload
             ) + self.outgoingCompression.flush(2)
+
+        wirePacket = self._createPacket(payload)
+
+        self.transport.write(wirePacket)
+        self.outgoingPacketSequence += 1
+
+    def _createPacket(self, payload: bytes) -> bytes:
+        """
+        Return the SSH packet at it should be sent to the peer over the wire.
+
+        `payload` is the actual data.
+        """
+        # Start by computing the required padding.
+        #
+        # Extra 1 byte for the padding length
+        payloadSize = len(payload) + 1
+        isETM = self.currentEncryptions.isOutMACETM()
+
+        if not isETM:
+            # For not ETM, the packet length is also included.
+            payloadSize += 4
+
         bs = self.currentEncryptions.encBlockSize
-        # 4 for the packet length and 1 for the padding length
-        totalSize = 5 + len(payload)
-        lenPad = bs - (totalSize % bs)
+        lenPad = bs - (payloadSize % bs)
         if lenPad < 4:
             lenPad = lenPad + bs
-        packet = (
-            struct.pack("!LB", totalSize + lenPad - 4, lenPad)
-            + payload
-            + randbytes.secureRandom(lenPad)
+
+        # Create the unencrypted packet.
```

Comment:
Shouldn't this packet construction live in the `currentEncryptions` attribute?  The *only* reference here to `self` other than `self.currentEncryptions` is to `self.outgoingPacketSequence`, so maybe the thing to do here is `return self.currentEncryptions.makePacket(payload, self.outgoingPacketSequence)` ?

(This is definitely a big improvement to at least get stuff into a functional shape, but pushing it a little further would put the responsibility in the right place.)

Reply:
> Shouldn't this packet construction live in the currentEncryptions attribute? 

Good question. This was my first refactoring.

And this is why I wanted an initial review for this draft PR.

Take a look at the first commit from the PR and let me know if you think that 
`return self.currentEncryptions.makePacket(payload, self.outgoingPacketSequence)` looks better


My initial refactoring was to move everything into SSHCiphers... but then, this means that SSHCiphers, beside actual cipher stuff it also need to know thinkgs about SSH transport and binary packet format.

This is why in the second version I moved the code to transport.

---------

In the end, I think that is best to keep this createPacket code into SSHTransport and not into SSHCipher